runtime/src/iree/base/tracing/console.c

```diff
@@ -120,6 +120,11 @@ static iree_zone_id_t iree_tracing_zone_begin(
   assert(_thread.depth + 1 < IREE_ARRAYSIZE(_thread.stack));
   iree_zone_id_t zone_id = ++_thread.depth;
 
+  // _console.file is not initialized, return.
+  if (!_console.file) {
+    return zone_id;
+  }
```

Comment:
> ConstEval::Runtime::Runtime() does not run IREE_TRACE_APP_ENTER()

Should we change that instead? More graceful error handling in the runtime code could still be useful though.

Another option is to just say that tracing the compiler only supports the `tracy` option in `IREE_TRACING_PROVIDER` not `console`. BTW, Tracy docs for profiling the compiler are here: https://iree.dev/developers/performance/profiling-with-tracy/#tracing-iree-compile . I don't think we have docs for the console tracing provider.

Reply:
I can add the `IREE_TRACE_APP_ENTER` and `IREE_TRACE_APP_EXIT` in the `-main.cc` files and add a warning message in `console.c`. Then the `IREE_TRACE_APP_*` in `Runtime.cpp` can be removed.